### PR TITLE
fix(tsdb): Add refs for file-sourced tag keys

### PR DESCRIPTION
**Pull request by benbjohnson**
This commit adds ref counting for files that we pull tag keys from. Previously, files were only ref counted during the time we extracted tag keys but this commit adds additional ref counting for the life of the `Engine.tagKeysNoPredicate()` function.

### CODE REVIEW

File: tsdb/tsm1/engine_schema.go [hunk 1]
```diff
@@ -358,6 +358,12 @@ func (e *Engine) tagKeysNoPredicate(ctx context.Context, orgID, bucketID influxd
 	var stats cursors.CursorStats
 	var canceled bool
 
+	// NOTE(bbj): Adding padding in case additional files are added between files
+	// instantiation & the ForEachFile(). The 'files' reference needs to be stable
+	// so defer can clean it up at the end.
+	files := make(unrefs, 0, e.FileStore.Count()+10)
+	defer files.Unref()
```
Comment (stuartcarnie) [inline tsdb/tsm1/engine_schema.go:365]: You can capture `files` in a closure, it won't matter if it grows:

https://play.golang.org/p/xYwL9_OT07S

Comment (benbjohnson) [reply tsdb/tsm1/engine_schema.go:365]: Ah, I like that better. 👍

**Comment by benbjohnson** [comment]
alchemy please merge

**Comment by benbjohnson** [comment]
I checked all the other `FileStore.ForEachFile()` uses and they all appear to be safely done so no other updates needed.

**Pull request merged**